models/m_payment_mollie_api.erl

```diff
@@ -313,52 +348,48 @@ handle_new_payment(PaymentId, Payment, #{ <<"recurringType">> := <<"recurring">>
                     ok = m_payment:update_psp_handler(NewPaymentId, PSPHandler, Context),
                     update_payment_status(NewPaymentId, Status, DateTime, Context);
                 {error, _} = Error ->
-                    lager:error("Payment PSP Mollie could not insert received recurring payment for payment #~p: ~p ~p", [PaymentId, Error, JSON]),
+                    lager:error("Payment PSP Mollie could not insert received recurring payment for first payment #~p: ~p ~p",
+                                [FirstPaymentId, Error, JSON]),
                     Error
             end
     end;
-handle_new_payment(PaymentId, Payment, #{ <<"recurringType">> := <<"first">> } = JSON, Context) ->
+handle_new_payment(FirstPaymentId, FirstPayment, #{ <<"sequenceType">> := <<"first">> } = JSON, Context) ->
     % First recurring payment for an existing payment - start the subscription
     #{
         <<"status">> := Status
     } = JSON,
     DateTime = z_convert:to_datetime( status_date(JSON) ),
-    case update_payment_status(PaymentId, Status, DateTime, Context) of
-        ok ->
-            case Status of
-                <<"paid">> -> create_subscription(Payment, Context);
-                _ -> ok
-            end;
-        {error, _} = Error ->
-            Error
+    case update_payment_status(FirstPaymentId, Status, DateTime, Context) of
+        ok when Status =:= <<"paid">> -> create_subscription(FirstPayment, Context);
+        ok -> ok;
+        {error, _} = Error -> Error
     end;
-handle_new_payment(PaymentId, _Payment, JSON, Context) ->
+handle_new_payment(OneOfPaymentId, _OneOfPayment, JSON, Context) ->
```

Comment:
Is this about specifically non-recurring payments? Then OneOf should be with "ff": one-off payment.

Reply:
Indeed.  Will update.